source/MaterialXGenShader/DefaultColorManagementSystem.cpp

```diff
@@ -16,20 +16,36 @@ const string DefaultColorManagementSystem::CMS_NAME = "default_cms";
 // DefaultColorManagementSystem methods
 //
 
-string DefaultColorManagementSystem::getImplementationName(const ColorSpaceTransform& transform) const
+ImplementationPtr DefaultColorManagementSystem::getImplementation(const ColorSpaceTransform& transform) const
 {
-    return "IM_" + transform.sourceSpace + "_to_" + transform.targetSpace + "_" + transform.type->getName() + "_" + _target;
+    if (!_document)
+    {
+        throw ExceptionShaderGenError("No library loaded for color management system");
+    }
+
+    TargetDefPtr targetDef = _document->getTargetDef(_target);
+    const StringVec targets = targetDef->getMatchingTargets();
```

Comment:
Find all targets inherited by this target, and search for implementations for all of them, returning the first one found. The  target list is already in order of priority.